expression/integration_test.go

```diff
@@ -6803,7 +6813,7 @@ func (s *testIntegrationSerialSuite) TestMixCollation(c *C) {
 	tk.MustQuery("select coercibility(concat(concat(mb4unicode, mb4general), concat(unicode, general))) from t;").Check(testkit.Rows("1"))
 	tk.MustQuery("select collation(concat(i, 1)) from t;").Check(testkit.Rows("utf8mb4_general_ci"))
 	tk.MustQuery("select coercibility(concat(i, 1)) from t;").Check(testkit.Rows("4"))
-	tk.MustQuery("select collation(concat(i, user())) from t;").Check(testkit.Rows("utf8mb4_general_ci"))
+	tk.MustQuery("select collation(concat(i, user())) from t;").Check(testkit.Rows("utf8mb4_bin"))
```

Comment:
Why change this result? I feel like I'm testing the following results on mysql, is it a process problem?
```
mysql> create table t (i int);
Query OK, 0 rows affected (0.04 sec)

mysql> insert into t values ( 1);
Query OK, 1 row affected (0.01 sec)

mysql>  set names utf8mb4 collate utf8mb4_general_ci;
Query OK, 0 rows affected (0.01 sec)

mysql> select collation(concat(i, user())) from t;
+------------------------------+
| collation(concat(i, user())) |
+------------------------------+
| utf8_general_ci              |
+------------------------------+
1 row in set (0.00 sec)
```

Reply:
it is a bug. MySQL result is actually the `utf8` character set and `utf8_general_ci` collation, see https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_user. and its coercibility is `system const` which is 3 [doc](https://dev.mysql.com/doc/refman/5.7/en/charset-collation-coercibility.html#:~:text=A%20%E2%80%9Csystem%20constant%E2%80%9D%20(the%20string%20returned%20by%20functions%20such%20as%20USER()%20or%20VERSION())%20has%20a%20coercibility%20of%203.)

so I decide use `utf8mb4` and its default collation `utf8mb4_bin`